src/test/java/org/prebid/server/bidder/madsense/MadsenseBidderTest.java

```diff
@@ -0,0 +1,399 @@
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorWhenFirstImpExtCouldNotBeParsed() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(imp -> imp
+                .banner(Banner.builder().build())
+                .ext(givenInvalidImpExt()));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = target.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1).containsExactly(badInput("Error parsing imp.ext parameters"));
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldCreateSeparateRequestForEachBannerImp() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(
+                builder -> builder.id("bannerImp1").banner(Banner.builder().build())
+                        .ext(givenImpExt("testCompanyId")),
+                builder -> builder.id("bannerImp2").banner(Banner.builder().build())
+                        .ext(givenImpExt("otherCompanyId")));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = target.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(2);
+        assertThat(result.getValue())
+                .extracting(HttpRequest::getImpIds, r -> r.getPayload().getImp().getFirst().getId())
+                .containsExactlyInAnyOrder(
+                        tuple(Set.of("bannerImp1"), "bannerImp1"),
+                        tuple(Set.of("bannerImp2"), "bannerImp2")
+                );
+        assertThat(result.getValue())
+                .extracting(HttpRequest::getUri)
+                .containsOnly(ENDPOINT_URL + "?company_id=testCompanyId");
+    }
+
+    @Test
+    public void makeHttpRequestsShouldCreateOneRequestForAllVideoImps() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(
+                builder -> builder.id("videoImp1").video(Video.builder().build())
+                        .ext(givenImpExt("testCompanyId")),
+                builder -> builder.id("videoImp2").video(Video.builder().build())
+                        .ext(givenImpExt("otherCompanyId")));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = target.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1);
+        assertThat(result.getValue().getFirst().getImpIds()).containsExactlyInAnyOrder("videoImp1", "videoImp2");
+        assertThat(result.getValue().getFirst().getPayload().getImp())
+                .extracting(Imp::getId)
+                .containsExactlyInAnyOrder("videoImp1", "videoImp2");
+        assertThat(result.getValue().getFirst().getUri())
+                .isEqualTo(ENDPOINT_URL + "?company_id=testCompanyId");
```

Comment:
Assert the endpoint in only two tests:
- one that expects `companyId` from `ext`
- other for `test=1` case